src/Ipn/Handler.php

```diff
@@ -197,19 +204,25 @@ protected function getPaymentEntity(array $ipn_data) {
    *   The data of the IPN request.
    * @param array $supported_types
    *   The IPN types to validate against.
+   * @param string $env
+   *   The BlueSnap environment of the payment gateway.
    *
    * @throws \Symfony\Component\HttpKernel\Exception\BadRequestHttpException
    *   If the IPN is not of a supported type.
    */
-  public function validateType(array $ipn_data, array $supported_types) {
+  public function validateType(array $ipn_data, array $supported_types, $env) {
     $ipn_type = $this->getType($ipn_data);
 
     if (!in_array($ipn_type, $supported_types)) {
-      $message = sprintf(
-        'IPN of type "%s" is not supported by the payment gateway.',
-        $ipn_type
-      );
-      $this->logger->error($message);
+      // Only log the message in non-production environments.
+      if ($env !== 'production') {
+        $message = sprintf(
+          'IPN of type "%s" is not supported by the payment gateway.',
+          $ipn_type
+        );
+        $this->logger->error($message);
+      }
```

Comment:
**CHANGE REQUEST**

Unneeded nesting:

```
$is_supported = in_array($ipn_type, $supported_types);
if ($is_supported && $env !== 'production') {
  //
}
```

**CHANGE REQUEST**

There are 2 concepts:

1. IPNs supported by the module.
2. IPNs supported by the gateway that handles the IPN.

The `validateType` function checks the 2nd case.
The `getGroupType` function checks the 1st case; please do the same for that as we'll get errors logged there as well.

Reply:
I made it nested because we still want to throw the error on all environments, right?

Reply:
@krystalcode I've made the changes to the getGroupType as well now.